src/Testcontainers/Configurations/WaitStrategies/HttpWaitStrategy.cs

```diff
@@ -75,51 +77,54 @@ public async Task<bool> UntilAsync(IContainer container)
             httpRequestMessage.Headers.Add(httpHeader.Key, httpHeader.Value);
           }
 
-          HttpResponseMessage httpResponseMessage;
-
-          try
-          {
-            httpResponseMessage = await httpClient.SendAsync(httpRequestMessage)
-              .ConfigureAwait(false);
-          }
-          catch (HttpRequestException)
-          {
-            return false;
-          }
-
-          Predicate<HttpStatusCode> predicate;
-
-          if (!_httpStatusCodes.Any() && _httpStatusCodePredicate == null)
+          using (httpRequestMessage.Content = _contentFactory())
```

Comment:
Do we need the `using` block here? I think disposing the HTTP client will dispose the message anyway, that was our initial issue, right?

Reply:
yes you are right... but according to the follwing blog, they changed the behaviour from .NET core 3.0 onwards, so that it will not be disposed implicitly anymore with newer .NET (core) versions: https://makolyte.com/csharp-disposing-the-request-httpcontent-when-using-httpclient/

Because this is a .NET standard project and we don't really know which version the consumer is running, I thought it is better to dispose it explicitly and live with it that it gets disposed twice in older versions.